locations/jclouds/src/main/java/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -204,7 +205,8 @@
     private static final Pattern LIST_PATTERN = Pattern.compile("^\\[(.*)\\]$");
     private static final Pattern INTEGER_PATTERN = Pattern.compile("^\\d*$");
 
-    private static boolean loggedSshKeysHint = false;
+    private final AtomicBoolean loggedSshKeysHint = new AtomicBoolean(false);
+    private final AtomicBoolean listedAvailableTemplatesOnNoSuchTemplate = new AtomicBoolean(false);
```

Comment:
Yes, I wasn't sure what to do to achieve the actual intent. `ThreadLocal` wouldn't work, maybe a `Map` with the location ID as the key? I wasn't sure whether the IDs would be the same for two `Location` instances generated by the management context for the same underlying location?